ansible_base/authentication/authenticator_plugins/google_oauth2.py

```diff
@@ -0,0 +1,116 @@
+    documentation_url = "https://python-social-auth.readthedocs.io/en/latest/backends/google.html#google-oauth2"
+
+    #################################
+    # Minimal params
+    #################################
+
+    KEY = CharField(
+        help_text=_("The OAuth2 key from your web application."),
+        allow_null=False,
+        ui_field_label=_('Google OAuth2 Key'),
+    )
+
+    SECRET = CharField(
+        help_text=_("The OAuth2 secret from your web application."),
+        allow_null=True,
+        ui_field_label=_('Google OAuth2 Secret'),
+    )
+
+    #################################
+    # Additional params
+    #################################
+
+    CALLBACK_URL = URLField(
+        help_text=_(
+            'Provide this URL as the callback URL for your application as part of your registration process. Refer to the documentation for more detail.'
+        ),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Callback URL"),
+    )
+
+    ACCESS_TOKEN_URL = URLField(
+        help_text=_("The URL to obtain an access token from the OIDC provider."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Access Token URL"),
+    )
+
+    ACCESS_TOKEN_METHOD = ChoiceField(
+        help_text=_("The HTTP method to be used when requesting an access token. Typically 'POST' or 'GET'."),
+        default="POST",
+        allow_null=True,
+        required=False,
+        choices=['GET', 'PUT', 'POST', 'PATCH', 'DELETE'],
+        ui_field_label=_("Access Token Method"),
+    )
+
+    AUTHORIZATION_URL = URLField(
+        help_text=_("The URL to redirect the user for Google OAuth2  provider authorization."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Authorization URL"),
+    )
+
+    REDIRECT_STATE = BooleanField(
+        help_text=_("Enable or disable state parameter in the redirect URI. Recommended to be True for preventing CSRF attacks."),
+        default=False,
+        allow_null=True,
+        ui_field_label=_("Redirect State"),
+    )
+
+    REVOKE_TOKEN_METHOD = ChoiceField(
+        help_text=_("The HTTP method to be used when revoking an access token. Typically 'POST' or 'GET'."),
+        default="GET",
+        allow_null=True,
+        choices=['GET', 'PUT', 'POST', 'PATCH', 'DELETE'],
+        ui_field_label=_("Revoke Token Method"),
+    )
+
+    REVOKE_TOKEN_URL = URLField(
+        help_text=_("The URL to revoke tokens. Used in the token revocation flow."),
+        required=False,
+        allow_null=True,
+        ui_field_label=_("Revoke Token URL"),
+    )
+
+    SCOPE = ListField(
+        help_text=_('The authorization scope for users. Defaults to ["openid", "email", "profile"].'),
+        required=False,
+        allow_null=False,
+        ui_field_label=_('List of OAuth2 Scope(s)'),
+        default=["openid", "email", "profile"],
+    )
+
+    EXTRA_DATA = ListField(
```

Comment:
Should we leverage the EXTRA_UNVERIFIED_FIELDS for this or is it different?

Reply:
What I find confusing is that in AWX is it `EXTRA_ARGUMENTS` https://github.com/ansible/awx/blob/b076cb00a968f2fe5b65cbe1ad8b328fd03b030c/awx/sso/conf.py#L619-L633 but in the backend code it's `EXTRA_DATA` https://github.com/python-social-auth/social-core/blob/master/social_core/backends/google.py#L73-L77

Reply:
The google oauth2 backend subclasses the base oauth2 backend where a comment about `EXTRA_DATA` resides ...

https://github.com/python-social-auth/social-core/blob/master/social_core/backends/oauth.py#L30-L32
```
    Settings will be inspected to get more values names that should be
    stored on extra_data field. The setting name is created following the
    pattern SOCIAL_AUTH_<uppercase current backend name>_EXTRA_DATA.
```